### ISSUE TITLE
Close and re-open ASP.NET Core Web Application - squiggles in HomeController.cs and warnings in NuGet dependencies tree

### ISSUE THREAD

**Issue by balachir** [dotnet/roslyn-project-system#1131]
## Steps to reproduce
Using VS 2017 RC.3 (Build d15rel 26109.02) with NetCore workload
New project ASP.NET Core Web Application (.NET Core) > 1.0 > Web Application (No Auth)
Close and re-open project

## Actual behavior
Squiggles in HomeController.cs
NuGet dependencies shows warnings

![image](https://cloud.githubusercontent.com/assets/8246794/21831964/9f008140-d75c-11e6-967c-7d341fc4b418.png)

## Expected behavior
No squiggles or warnings

**Comment by davkean** [comment]
I'm also seeing this with 26107.01 D15PreRel. Looks like maybe the design-time build is failing?

**Comment by mavasani** [comment]
I am unable to repro this on a clean install of D15PreRel 26111.7

1. File > New Project > ASP.NET Core Web App (.Net Core)
2. ASP.NET Core Templates 1.0 > Web Application > OK
3. Open Controllers > HomeController.cs - squiggles show while restore is happening in background.
4. Once background restore completes, squiggles go away.
5. Close and reopen solution - verified no squiggles and NuGet dependencies also show up correctly in the solution explorer
6. Close and reopen VS and open solution - everything works fine.

**Comment by 333fred** [comment]
I'm seeing this issue on the d15rel, 26111.01

**Comment by 333fred** [comment]
Hmm. The above was on my dev box. On a clean VM with 26112.00, I can't repro. I'll try updating my dev box.

**Comment by davkean** [comment]
Design time build is probably broken - is this caused by us failing to load an assembly System.IO.FileSystem? @natidea

**Comment by davkean** [comment]
It's broken on my box and I get the followign in the Error List (sometimes)

```
Severity	Code	Description	Project	File	Line	Suppression State
Error	MSB4018	The "ResolvePackageDependencies" task failed unexpectedly.
System.IO.FileNotFoundException: Could not load file or assembly 'System.IO.FileSystem, Version=4.0.1.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a' or one of its dependencies. The system cannot find the file specified.
File name: 'System.IO.FileSystem, Version=4.0.1.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a'
   at Microsoft.NET.Build.Tasks.LockFileCache.LoadLockFile(String path)
   at Microsoft.NET.Build.Tasks.LockFileCache.GetLockFile(String path)
   at Microsoft.NET.Build.Tasks.ResolvePackageDependencies.get_LockFile()
   at Microsoft.NET.Build.Tasks.ResolvePackageDependencies.ReadProjectFileDependencies()
   at Microsoft.NET.Build.Tasks.ResolvePackageDependencies.ExecuteCore()
   at Microsoft.NET.Build.Tasks.TaskBase.Execute()
   at Microsoft.Build.BackEnd.TaskExecutionHost.Microsoft.Build.BackEnd.ITaskExecutionHost.Execute()
   at Microsoft.Build.BackEnd.TaskBuilder.<ExecuteInstantiatedTask>d__26.MoveNext()

WRN: Assembly binding logging is turned OFF.
To enable assembly bind failure logging, set the registry value [HKLM\Software\Microsoft\Fusion!EnableLog] (DWORD) to 1.
Note: There is some performance penalty associated with assembly bind failure logging.
To turn this feature off, remove the registry value [HKLM\Software\Microsoft\Fusion!EnableLog].
	ConsoleApp41 (loading...)	C:\Program Files (x86)\Microsoft Visual Studio\2017\Enterprise\MSBuild\Sdks\Microsoft.NET.Sdk\build\Microsoft.PackageDependencyResolution.targets	154	

```

Doesn't look like we're always logging design-time build errors.

**Comment by 333fred** [comment]
From my testing, it's looking like the initial restore isn't being run. I'm not getting a `project.assets.json` in the obj folder. If I run a restore from the cli, or if I add a package to the solution, a restore occurs correctly and the squiggles go away.

**Comment by 333fred** [comment]
We are calling [`NominateProjectAsync`](https://github.com/dotnet/roslyn-project-system/blob/master/src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/NuGet/NuGetRestorer.cs#L140), however it seems to no be honored.

**Comment by natidea** [comment]
@davkean Newer builds have `C:\Program Files (x86)\Microsoft Visual Studio\2017\Enterprise\MSBuild\Sdks\Microsoft.NET.Sdk\tools\net46\System.IO.FileSystem.dll` which should prevent your error

**Comment by mavasani** [comment]
@natidea do we have a bug for the System.IO.FileSystem issue - this one should be duped to that.

**Comment by davkean** [comment]
It's not the same bug, Fred's running into a different issue where NuGet isn't restoring.

**Comment by 333fred** [comment]
I've narrowed behavior futher: NuGet doesn't seem to perform the first restore after a solution is opened. An example:
1. Create a console app. No restore is performed. When I manually trigger a restore, I see `All packages are already installed and there is nothing to restore.` in the package manager log.
2. Add a NuGet package. Restore occurs, and intellisense appears for all packages.
3. Close the solution and add a new package.
4. Reopen the solution. The first two packages are resolved and have intellisense, the 3rd package isn't restored.
5. Add another package. All 4 packages will be restored.

**Comment by 333fred** [comment]
Currently engaging with the NuGet team; they have a bug that is likely the cause of this that should have been fixed in 26112.00. I can't repro with a clean build, but I can with an upgrade from 26111.01. Working to figure out why.

**Comment by 333fred** [comment]
This no longer repros on current builds, and I could not repro the issue when upgrading an RC2 build to the latest internal builds.
